lang_GENERIC_base/AST_generic.ml

```diff
@@ -842,6 +844,9 @@ and for_header =
                expr (* pattern 'in' expr *)
   (* sgrep: *)
   | ForEllipsis of tok (* ... *)
+  (* Lua *)
+  | ForIn of for_var_or_expr list (* init *) *
```

Comment:
Python and Go have this 'for pattern in expr', is there a need for adding another constructor?